### PR TITLE
Update analytics.tsx

### PR DESCRIPTION
**Pull request by martyna-mindsdb**

### CODE REVIEW

File: packages/blog-starter-kit/themes/enterprise/next.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const ANALYTICS_BASE_URL = 'https://hn-ping2.hashnode.com';
-const ADVANCED_ANALYTICS_BASE_URL = 'https://stats.hashnode.com';
+const ADVANCED_ANALYTICS_BASE_URL = 'https://user-analytics.hashnode.com';
```
Comment (JannikWempe) [inline packages/blog-starter-kit/themes/enterprise/next.config.js:2]: This is currently used for proxying `/api/collect` (https://github.com/mindsdb/hashnode-starter-kit/pull/17/files#diff-a9b0cf02522cd2a8797011d00973b99b76dc5ce6b23570c6731a3d38d3d7b211R40-R41) but the new analytics calls are made to `/api/analytics`.

This
```javascript
{
  source: '/api/collect',
  destination: `${ADVANCED_ANALYTICS_BASE_URL}/api/collect`,
}
```
should be changed to 
```javascript
{
  source: '/api/analytics,
  destination: `${ADVANCED_ANALYTICS_BASE_URL}/api/analytics`,
}
```

Comment (martyna-mindsdb) [reply packages/blog-starter-kit/themes/enterprise/next.config.js:2]: Thank you @JannikWempe 
I made this update. Please let me know if it is fine now.

Comment (JannikWempe) [reply packages/blog-starter-kit/themes/enterprise/next.config.js:2]: This part is fixed now. I have cloned the repo to check if everything is working now but I have found several more issues:

- The new `useEffect` is called conditionally due to the `isProd` check ([see here](https://github.com/mindsdb/hashnode-starter-kit/pull/17/files#diff-049e1536cc838d860847f370d4507d0a86e4cb72ba91887741f73d3d4ffcc3ffR21)). This is violating the rules of React hooks.
- You now have 2 `useEffect`s and 3 of 4 calls are made in both of them.

This is the [PR in our starter kit](https://github.com/Hashnode/starter-kit/commit/f543e14caf8cf017416638d8d60a3e1e3a0c1de1#diff-049e1536cc838d860847f370d4507d0a86e4cb72ba91887741f73d3d4ffcc3ff) that introduced the new analytics. You can take it as a reference.

File: packages/blog-starter-kit/themes/enterprise/components/analytics.tsx [hunk 2]
```diff
@@ -138,5 +117,91 @@ export const Analytics = () => {
 		});
 	};
 
+	function _sendViewsToAdvancedAnalyticsDashboard() {
+		const publicationId = publication.id;
+		const postId = post && post.id;
+		const seriesId = series?.id || post?.series?.id;
+		const staticPageId = page && page.id;
```
Comment (JannikWempe) [inline packages/blog-starter-kit/themes/enterprise/components/analytics.tsx:124]: `series` and `page` are not defined. They have to be passed as a prop to this `Analytics` component.

Check [this commit in our starter kit](https://github.com/Hashnode/starter-kit/commit/f543e14caf8cf017416638d8d60a3e1e3a0c1de1#diff-049e1536cc838d860847f370d4507d0a86e4cb72ba91887741f73d3d4ffcc3ff) that made the change.

Comment (martyna-mindsdb) [reply packages/blog-starter-kit/themes/enterprise/components/analytics.tsx:105]: I updated the analytics.tsx file to be exactly as in the commit you linked.

Comment (JannikWempe) [reply packages/blog-starter-kit/themes/enterprise/components/analytics.tsx:105]: There are some changes missing outside of this file.

I have created a PR with the missing required changes. You can either merge it or do the changes on this branch if you want.

https://github.com/mindsdb/hashnode-starter-kit/pull/19/files

Comment (martyna-mindsdb) [reply packages/blog-starter-kit/themes/enterprise/components/analytics.tsx:105]: Thanks @JannikWempe 
I'll close this PR and merge the one you created.

Comment (JannikWempe) [reply packages/blog-starter-kit/themes/enterprise/components/analytics.tsx:105]: You are welcome 😊
You can tag me if you have a question or need help.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**